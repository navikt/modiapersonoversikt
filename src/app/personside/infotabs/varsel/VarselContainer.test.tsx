import * as React from 'react';
import * as renderer from 'react-test-renderer';
import TestProvider from '../../../../test/Testprovider';
import VarselContainer from './VarslerContainer';
import { setupReactQueryMocks } from '../../../../test/testStore';

test('Viser varselcontainer med alt innhold', () => {
    setupReactQueryMocks();
    const container = renderer.create(
        <TestProvider>
            <VarselContainer />
        </TestProvider>
    );

    const json = container.toJSON();
    expect(json).toMatchSnapshot();
});
