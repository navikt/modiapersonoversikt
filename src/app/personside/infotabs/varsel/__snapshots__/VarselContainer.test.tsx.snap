// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viser varselcontainer med alt innhold 1`] = `
.c5 {
  font-weight: bold;
}

.c9 {
  padding: 0.7rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c9 > *:first-child {
  -webkit-flex: 0 0 20%;
  -ms-flex: 0 0 20%;
  flex: 0 0 20%;
}

.c9 > *:last-child {
  max-width: 30rem;
}

.c8 {
  border-radius: 0.5rem;
  background-color: #f1f0f0;
  box-shadow: inset 0 0 0 0.3rem white,inset 0 0 0 0.37rem rgba(0,0,0,0.15);
  padding: 1.25rem;
}

.c8 > *:not(:first-child) {
  border-top: solid 0.0625rem #78706a;
}

.c7 {
  border: none;
  cursor: pointer;
  background-color: transparent;
  padding: 0.1rem;
  border-radius: 0.5em;
}

.c7:focus {
  outline: none;
  box-shadow: 0 0 0 0.1875rem #254b6d;
}

.c7:focus {
  outline: none;
  box-shadow: none;
}

.c7:focus::after {
  position: absolute;
  top: 0;
  left: 0;
  content: '';
  box-shadow: inset 0 0 0 0.1875rem #254b6d;
  height: 100%;
  width: 100%;
  pointer-events: none;
}

.c3 {
  border-radius: 0.25rem;
  background-color: white;
  box-shadow: 0 0 0.5rem rgba(0,0,0,0.1),0 0 0 1px rgba(0,0,0,0.1);
}

.c4 {
  position: relative;
  display: -ms-grid;
  display: grid;
  -ms-grid-columns: 6rem 55% 1fr auto;
  grid-template-columns: 6rem minmax(35%,55%) 1fr auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: .25rem;
  cursor: pointer;
}

.c4 > *:nth-child(1) {
  -ms-grid-column: 1;
}

.c4 > *:nth-child(2) {
  -ms-grid-column: 2;
}

.c4 > *:nth-child(3) {
  -ms-grid-column: 3;
}

.c4 > *:nth-child(4) {
  -ms-grid-column: 4;
}

.c4:hover {
  box-shadow: inset 0 0 0 0.1rem #0067c5,0 0.1rem 0.1rem rgba(0,0,0,0.2);
}

.c4 > * {
  padding: 0.7rem;
}

.c6 li {
  display: inline-block;
}

.c6 li:not(:last-child):after {
  content: ',';
  margin-right: 0.5em;
}

.c0 {
  padding: .8rem;
}

.c1 {
  display: -ms-grid;
  display: grid;
  -ms-grid-columns: 6rem 55% 1fr;
  grid-template-columns: 6rem minmax(35%,55%) 1fr;
  font-weight: bold;
}

.c1 > *:nth-child(1) {
  -ms-grid-column: 1;
}

.c1 > *:nth-child(2) {
  -ms-grid-column: 2;
}

.c1 > *:nth-child(3) {
  -ms-grid-column: 3;
}

.c1 > * {
  padding: 0.7rem 0.7rem 0.3rem;
}

.c2 > * {
  margin-top: 0.5rem;
}

<article
  className="c0"
>
  <div
    className="c1"
  >
    <p>
      Dato
    </p>
    <p>
      Type
    </p>
    <p>
      Kanal
    </p>
  </div>
  <ol
    aria-label="Brukerens varsler"
    className="c2"
  >
    <li
      aria-label="Påminnelse om å sende meldekort"
      className="c3"
    >
      <div
        className="c4"
        onClick={[Function]}
      >
        <p
          className="typo-normal"
        >
          01.01.1970
        </p>
        <p
          className="typo-normal"
        >
          <span
            className="c5"
          >
            Påminnelse om å sende meldekort
          </span>
        </p>
        <ul
          aria-label="Kommunikasjonskanaler"
          className="c6"
        >
          <li
            className="typo-normal"
          >
            EPOST
          </li>
          <li
            className="typo-normal"
          >
            SMS
          </li>
        </ul>
        <button
          aria-expanded={false}
          aria-label="Vis meldinger"
          className="c7"
          onClick={[Function]}
          title="Vis meldinger"
        >
          <i
            className="nav-frontend-chevron chevronboks chevron--ned"
          />
        </button>
      </div>
      <ol
        aria-label="Meldinger"
        className="c8"
      >
        <li
          aria-label="EPOST"
          className="c9"
        >
          <div>
            <p
              className="typo-normal"
            >
              <span
                className="c5"
              >
                EPOST
              </span>
            </p>
            <p
              className="typo-normal"
            >
              01. jan. 1970
            </p>
          </div>
          <div>
            <div>
              <p
                className="typo-normal"
              >
                <span
                  className="c5"
                >
                  Epostemne
                </span>
              </p>
              <p
                className="typo-normal"
              >
                Meldingsinnhold
              </p>
            </div>
            <p
              className="typo-normal"
            >
              <span
                className="c5"
              >
                Epost: 
                mottakerinfo
              </span>
            </p>
          </div>
        </li>
        <li
          aria-label="SMS"
          className="c9"
        >
          <div>
            <p
              className="typo-normal"
            >
              <span
                className="c5"
              >
                SMS
              </span>
            </p>
            <p
              className="typo-normal"
            >
              01. jan. 1970
            </p>
          </div>
          <div>
            <p
              className="typo-normal"
            >
              Meldingsinnhold
            </p>
            <p
              className="typo-normal"
            >
              <span
                className="c5"
              >
                Tlf: 
                mottakerinfo
              </span>
            </p>
          </div>
        </li>
      </ol>
    </li>
    <li
      aria-label="Påminnelse om å sende meldekort"
      className="c3"
    >
      <div
        className="c4"
        onClick={[Function]}
      >
        <p
          className="typo-normal"
        >
          01.01.1970
        </p>
        <p
          className="typo-normal"
        >
          <span
            className="c5"
          >
            Påminnelse om å sende meldekort
          </span>
        </p>
        <ul
          aria-label="Kommunikasjonskanaler"
          className="c6"
        >
          <li
            className="typo-normal"
          >
            EPOST
          </li>
        </ul>
        <button
          aria-expanded={false}
          aria-label="Vis meldinger"
          className="c7"
          onClick={[Function]}
          title="Vis meldinger"
        >
          <i
            className="nav-frontend-chevron chevronboks chevron--ned"
          />
        </button>
      </div>
      <ol
        aria-label="Meldinger"
        className="c8"
      >
        <li
          aria-label="EPOST"
          className="c9"
        >
          <div>
            <p
              className="typo-normal"
            >
              <span
                className="c5"
              >
                EPOST
              </span>
            </p>
            <p
              className="typo-normal"
            >
              01. jan. 1970
            </p>
          </div>
          <div>
            <div>
              <p
                className="typo-normal"
              >
                <span
                  className="c5"
                >
                  Epostemne
                </span>
              </p>
              <p
                className="typo-normal"
              >
                Meldingsinnhold
              </p>
            </div>
            <p
              className="typo-normal"
            >
              <span
                className="c5"
              >
                Epost: 
                mottakerinfo
              </span>
            </p>
          </div>
        </li>
      </ol>
    </li>
  </ol>
</article>
`;
